tests/unit/utils_test/test_data_utils.py

```diff
@@ -3,9 +3,36 @@
 
 import numpy as np
 from numpy.testing import assert_array_equal
-import zarr
+
+try:
+    import zarr
+    ZARR_INSTALLED = True
+except ImportError:
+    ZARR_INSTALLED = False
+
+
+class MyIterable:
+    def __init__(self, data):
+        self.data = data
+
 
 class TestAppendData(TestCase):
+    def test_append_exception(self):
+        data = MyIterable([1, 2, 3, 4, 5])
+        with self.assertRaises(ValueError):
+            append_data(data, 4)
+
+
+class TestZarrAppendData(TestCase):
+
+    def setUp(self):
+        if not ZARR_INSTALLED:
+            self.skipTest("optional Zarr package is not installed")
+
+    def test_append_exception(self):
+        data = MyIterable([1, 2, 3, 4, 5])
+        with self.assertRaises(ValueError):
+            append_data(data, 4)
```

Comment:
```suggestion

```

If you consolidate the error handling I think you would not need the same append exception test twice